### PR TITLE
Dynamic rancher instance

### PR DESCRIPTION
**Pull request by igomez06**
This adds a new corral package where the Kubernetes version and the rancher version can by dynamically defined. This would be used for QA automation.

### CODE REVIEW

File: templates/aws/cluster_nodes/terraform/pools/main.tf [hunk 1]
```diff
@@ -147,20 +161,48 @@ resource "aws_lb_target_group_attachment" "aws_internal_tg_attachment_443_server
...
-  count = var.agent_count
+  count = var.airgap_setup ? var.agent_count : 0
   target_group_arn = aws_lb_target_group.aws_tg_80.arn
   target_id        = aws_instance.agent[count.index].id
   port             = 80
 }
 
 resource "aws_lb_target_group_attachment" "aws_tg_attachment_443" {
-  count = var.agent_count
+  count = var.airgap_setup ? var.agent_count : 0
```
Comment (slickwarren) [inline templates/aws/cluster_nodes/terraform/pools/main.tf:186]: Don't we want a target group for this, even when airgap is not enabled? same comment for above tg_80

File: templates/rancher-airgap/overlay/opt/corral/rancher/install-rancher.sh [hunk 2]
```diff
@@ -4,20 +4,18 @@ set -ex
...
 
-CORRAL_rancher_host=${CORRAL_rancher_host:="${CORRAL_fqdn}"}
+CORRAL_internal_rancher_host=${CORRAL_internal_rancher_host:="${CORRAL_internal_fqdn}"}
 CORRAL_rancher_version=${CORRAL_rancher_version:=$(helm search repo rancher-latest/rancher -o json | jq -r .[0].version)}
 
 helm upgrade \
 --install \
 --create-namespace \
---set hostname="$CORRAL_rancher_host" \
+--set hostname="$CORRAL_internal_rancher_host" \
```
Comment (slickwarren) [inline templates/rancher-airgap/overlay/opt/corral/rancher/install-rancher.sh:13]: I think this is switched to avoid having to set the rancher URL later?

**Review by slickwarren — COMMENTED** [review]
overall lgtm, just have some clarifying asks

File: templates/aws/cluster_nodes/terraform/pools/main.tf (hunk 1 above)
Comment (igomez06) [reply templates/aws/cluster_nodes/terraform/pools/main.tf:186]: Sorry for the confusion this PR is based of this one: https://github.com/rancherlabs/corral-packages/pull/10
 can you write the comments there?

**Review by izaac — APPROVED** [review]

File: packages/aws/rancher-custom-cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  description: rancher-custom-cluster
+  variables:
+    node_count:
+      default: 1
+templates:
+  - aws/nodes
+  - rancher-custom-cluster
+variables:
+  airgap_setup:
+    - true
```
[the remark this answers is not in the document]
Comment (MSpencer87) [reply packages/aws/rancher-custom-cluster.yaml:12]: I should be able to reuse this for proxy, lets keep the general naming for now

**Comment by MSpencer87** [comment]
LGTM

**Review by MSpencer87 — APPROVED** [review]
LGTM

**Pull request merged**